artemis/db.py

```diff
@@ -376,6 +409,67 @@ def _get_task_deduplication_data(self, task: Task) -> str:
             }
         ).replace("\x00", " ")
 
+    def take_single_report_generation_task(self) -> Optional[ReportGenerationTask]:
+        with self.session() as session:
+            query = session.query(ReportGenerationTask).filter(
+                ReportGenerationTask.status == ReportGenerationTaskStatus.PENDING.value
+            )
+            if query.count() > 0:
```

Comment:
you can use return `query.first()`, as it gives either first row or None when query returns no row